### PR TITLE
Add loading state to Button component

### LINKED ISSUE #635: [Button] Add loading state
**Is your feature request related to a problem? Please describe.**
Button with loading state would be needed for the bag-transfer app
Screenshot 2019-04-04 at 17 15 34

### PR DESCRIPTION
**Pull request by KarinaDavtyan**
By passing `isLoading` prop Buttons state changes to loading.
Also added `color` prop in Loader component as it is required in design

closes #635

### CODE REVIEW

File: packages/universal-components/src/Button/ButtonInner.js [hunk 1]
```diff
@@ -78,24 +81,40 @@ export default function ButtonInner({
...
-        {leftIcon != null && <View style={rightSpace}>{leftIcon}</View>}
-        {children ||
-          (label != null && (
-            <ButtonTitle text={label} type={type} size={size} />
-          ))}
-      </View>
-      <View style={layout.row}>
-        {sublabel != null && (
+      {isLoading ? (
+        <View style={{ alignSelf: 'center' }}>
```
Comment (tbergq) [inline packages/universal-components/src/Button/ButtonInner.js]: Please create as style with StyleSheet.create

File: packages/universal-components/src/Button/ButtonInner.js [hunk 2]
```diff
@@ -78,24 +81,40 @@ export default function ButtonInner({
...
+                : defaultTokens.paletteWhite
+            }
+          />
+        </View>
+      ) : (
+        <>
           <View style={layout.row}>
-            <ButtonTitle text={sublabel} type={type} variant="sublabel" />
+            {leftIcon != null && <View style={rightSpace}>{leftIcon}</View>}
+            {children ||
```
Comment (tbergq) [inline packages/universal-components/src/Button/ButtonInner.js]: `children != null`

File: packages/universal-components/src/Button/ButtonInner.js (hunk 1 above)
Comment (KarinaDavtyan) [reply packages/universal-components/src/Button/ButtonInner.js]: Sorry, forgot to remove it, its not doing anything

File: packages/universal-components/src/Button/ButtonInner.js (hunk 2 above)
Comment (KarinaDavtyan) [reply packages/universal-components/src/Button/ButtonInner.js]: Where do you want to add this condition?

Comment (tbergq) [reply packages/universal-components/src/Button/ButtonInner.js]: Oh, I think I misread it, nevermind.

File: packages/universal-components/src/Button/ButtonInner.js (hunk 1 above)
Comment (KarinaDavtyan) [reply packages/universal-components/src/Button/ButtonInner.js]: Removed

File: packages/universal-components/src/Button/ButtonInner.js [hunk 3]
```diff
@@ -78,24 +81,38 @@ export default function ButtonInner({
...
-      <View style={layout.row}>
-        {leftIcon != null && <View style={rightSpace}>{leftIcon}</View>}
-        {children ||
-          (label != null && (
-            <ButtonTitle text={label} type={type} size={size} />
-          ))}
-      </View>
-      <View style={layout.row}>
-        {sublabel != null && (
+      {isLoading ? (
```
Comment (tbergq) [inline packages/universal-components/src/Button/ButtonInner.js]: What about passing this props to loader and from Loader to `ActivityIndicator` it has the animating props, which will hide the loader if false. [ActivityIndicator](https://facebook.github.io/react-native/docs/activityindicator#animating)

File: packages/universal-components/src/Button/GenericButtonWrapper.js [hunk 4]
```diff
@@ -26,12 +27,13 @@ export default function Button({
...
   testID,
   children,
+  isLoading,
 }: Props) {
   const onPressHandler = onPress || (() => {});
 
   return (
     <Touchable
-      disabled={!onPress || disabled}
+      disabled={!onPress || disabled || isLoading}
```
Comment (JoseDuSV) [inline packages/universal-components/src/Button/GenericButtonWrapper.js]: Web button is still enabled during loading. We need to add some similar condition like this native one to `Button.web.js` for both button types ( `<a>` & `<button>` ).

File: packages/universal-components/src/Button/ButtonInner.js (hunk 3 above)
Comment (KarinaDavtyan) [reply packages/universal-components/src/Button/ButtonInner.js]: Just pushed the changes. Did you mean this?

Comment (tbergq) [reply packages/universal-components/src/Button/ButtonInner.js]: Yes, looks better, now we got rid of the `isLoading ? ... : ...` from ButtonInner, so it is much more readable 🎉

File: packages/universal-components/src/Button/GenericButtonWrapper.js (hunk 4 above)
Comment (KarinaDavtyan) [reply packages/universal-components/src/Button/GenericButtonWrapper.js]: I ve changed this condition Button.web.js as well as Button.native.js. And removed one from GenericButtonWrapper as its not added anymore.
But random test errors appear so the check wont pass

**Comment by tbergq** [comment]
@KarinaDavtyan some tests are unhappy 😢

**Comment by KarinaDavtyan** [comment]
> @KarinaDavtyan some tests are unhappy 😢

@tbergq I know, there are random test errors appearing and I cannot update snapshot 😞
Screenshot 2019-04-05 at 14 55 03

File: packages/universal-components/src/Button/GenericButtonWrapper.js (hunk 4 above)
Comment (JoseDuSV) [reply packages/universal-components/src/Button/GenericButtonWrapper.js]: 👍 Thanks

Weird that tests are ok when I run it locally.

**Comment by tbergq** [comment]
@KarinaDavtyan, this bug is related to node version `11.11` please upgrade to latest version, or downgrade

**Pull request merged**